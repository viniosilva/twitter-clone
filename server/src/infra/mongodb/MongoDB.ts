import { Connection, createConnection } from 'mongoose';

export default class MongoDB {
  readonly conn: Connection;

  constructor(url: string, dbName: string) {
    this.conn = createConnection(url, {
      dbName,
      useNewUrlParser: true,
      useUnifiedTopology: true,
    });
  }

  async disconnect(): Promise<void> {
    await this.conn.close();
  }
}
