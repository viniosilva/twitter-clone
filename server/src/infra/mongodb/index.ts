import { mongodbConfig } from '../../AppConfig';
import MongoDB from './MongoDB';

export default new MongoDB(mongodbConfig.url, mongodbConfig.dbName);
