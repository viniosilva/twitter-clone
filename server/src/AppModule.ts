import { Module } from '@nestjs/common';
import { LoggerModule } from 'nestjs-pino';
import { httpLogger } from './infra/logger';
import AuthModule from './presentation/auth/AuthModule';
import HealthModule from './presentation/health/HealthModule';
import TweetModule from './presentation/tweet/TweetModule';

@Module({
  imports: [LoggerModule.forRoot({ pinoHttp: httpLogger }), HealthModule, AuthModule, TweetModule],
})
export default class AppModule {}
