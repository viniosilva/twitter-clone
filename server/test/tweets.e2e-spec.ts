import { Test, TestingModule } from '@nestjs/testing';
import { INestApplication } from '@nestjs/common';
import request, { Response } from 'supertest';
import AppModule from '../src/AppModule';
import mongodb from '../src/infra/mongodb';
import UserModel from '../src/domain/user/UserModel';

describe('TweetController (e2e)', () => {
  const path = '/api/tweets';
  const registerPath = '/api/auth/register';
  const loginPath = '/api/auth/login';

  let app: INestApplication;

  beforeEach(async () => {
    const moduleFixture: TestingModule = await Test.createTestingModule({
      imports: [AppModule],
    }).compile();

    app = moduleFixture.createNestApplication();
    await app.init();
  });

  afterEach(async () => {
    await UserModel.deleteMany();
  });

  afterAll(async () => {
    await mongodb.disconnect();
  });

  describe('POST /api/tweets', () => {
    it('should return the tweet', async () => {
      const registerPayload = { email: 'tweet@test.com', password: 'secret' };
      await request(app.getHttpServer())
        .post(registerPath)
        .send(registerPayload);

      const response = await request(app.getHttpServer())
        .post(loginPath)
        .send(registerPayload);

      const token = response.body.token;

      const payload = { content: 'TEST' };
      return request(app.getHttpServer())
        .post(path)
        .set('Authorization', `Bearer ${token}`)
        .send(payload)
        .expect(201)
        .expect((res: Response) => {
          expect(res.body).toMatchSnapshot({ createdAt: expect.any(String) });
        });
    });

    it('should throw forbiden exception when authorization is empty', () => {
      const payload = { content: 'TEST' };
      return request(app.getHttpServer())
        .post(path)
        .send(payload)
        .expect(403)
        .expect((res: Response) => {
          expect(res.body).toMatchSnapshot();
        });
    });
  });
});
