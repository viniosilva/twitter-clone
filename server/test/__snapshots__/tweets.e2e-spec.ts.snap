// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TweetController (e2e) POST /api/tweets should return the tweet 1`] = `
Object {
  "_id": 1,
  "content": "TEST",
  "createdAt": Any<String>,
  "likes": Array [],
}
`;

exports[`TweetController (e2e) POST /api/tweets should throw forbiden exception when authorization is empty 1`] = `
Object {
  "error": "Forbidden",
  "message": "Forbidden resource",
  "statusCode": 403,
}
`;
